zk_toolbox/crates/common/src/withdraw.rs

```diff
@@ -0,0 +1,305 @@
+use async_trait::async_trait;
+// use serde_json::Value;
+use ethers::{
+    providers::{JsonRpcClient, Provider, ProviderError},
+    types::Bytes,
+    types::{Address, TransactionReceipt, TxHash, H160, H256, U64},
+};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use zksync_system_constants::L1_MESSENGER_ADDRESS;
+use zksync_types::ethabi;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Proof {
+    pub id: u64,
+    #[serde(rename = "proof")]
+    pub merkle_proof: Vec<H256>,
+    pub root: Bytes,
+}
+
+/// Represents a L2 to L1 transaction log.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct L2ToL1Log {
```

Comment:
is it the same struct as in `core/lib/types`?

Reply:
Doesn't look the same: 
https://github.com/matter-labs/zksync-era/blob/8f43a5f0faeed29e8a3dde315bf612c7d22c1444/core/lib/types/src/l2_to_l1_log.rs#L11

Reply:
I found the one, there were two: 
https://github.com/matter-labs/zksync-era/blob/8f43a5f0faeed29e8a3dde315bf612c7d22c1444/core/lib/types/src/api/mod.rs#L190